$NetBSD: patch-makefile,v 1.3 2020/02/18 16:49:50 joerg Exp $

Exclude -flto for gcc 4.1.3 (used in NetBSD/vax 6)
Exclude -flto for clang on NetBSD
Exclude -flto for gcc 4.4.7 (Used in DragonFly 2.13 - 3.1)
Make -O2 conditional so can be overridden by environment
Fix SDL2 detection

--- makefile.orig	2018-09-20 05:36:48.-00145232 +0000
+++ makefile
@@ -217,7 +217,7 @@ ifeq ($(WIN32),)  #*nix Environments (&&
       $(shell git log -1 --pretty="SIM_GIT_COMMIT_ID %H%nSIM_GIT_COMMIT_TIME $(isodate)" >.git-commit-id)
     endif
   endif
-  LTO_EXCLUDE_VERSIONS = 
+  LTO_EXCLUDE_VERSIONS = 4.1.3 4.4.7 4.8.3 4.8.4 4.8.5 5.3.0 5.4.0
   PCAPLIB = pcap
   ifeq (agcc,$(findstring agcc,$(GCC))) # Android target build?
     OS_CCDEFS = -D_GNU_SOURCE
@@ -1039,20 +1039,23 @@ ifneq (,$(UNSUPPORTED_BUILD))
 endif
 ifneq ($(DEBUG),)
   CFLAGS_G = -g -ggdb -g3
-  CFLAGS_O = -O0
+  CFLAGS_O ?= -O0
   BUILD_FEATURES = - debugging support
 else
   ifneq (,$(findstring clang,$(COMPILER_NAME))$(findstring LLVM,$(COMPILER_NAME)))
-    CFLAGS_O = -O0
+    CFLAGS_O ?= -O0
     ifeq (Darwin,$(OSTYPE))
       NO_LTO = 1
+    endif
+    ifeq (NetBSD,$(OSTYPE))
+      NO_LTO = 1
     endif
   else
     NO_LTO = 1
     ifeq (Darwin,$(OSTYPE))
       CFLAGS_O += -O4 -fno-strict-overflow -flto -fwhole-program
     else
-      CFLAGS_O := -O2 -fno-strict-overflow 
+      CFLAGS_O ?= -O2 -fno-strict-overflow 
     endif
   endif
   LDFLAGS_O = 
